src/frontend/js/modules/table-of-content.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * Generate dynamic table of content
+ */
+export default class TableOfContent {
+  /**
+   * Initialize table of content
+   *
+   * @param {string} tagSelector - selector for tags to observe
+   * @param {string} tocWrapperSelector - selector for table of content wrapper
```

Comment:
better name it `parentElement`, because it actually is not a Table of Content wrapper. 

Table of Content wrapper is the `<div class="table-of-content"></div>`